lib/android/app/src/main/java/com/reactnativenavigation/views/stack/topbar/TopBar.java

```diff
@@ -123,12 +123,28 @@ public void setHeight(int height) {
         setLayoutParams(lp);
     }
 
+
     public void setTitleHeight(int height) {
-        titleAndButtonsContainer.setHeight(height);
+        int pixelHeight = UiUtils.dpToPx(getContext(), height);
+        ViewGroup.LayoutParams layoutParams = titleAndButtonsContainer.getLayoutParams();
+        if (pixelHeight == layoutParams.height) return;
+        layoutParams.height = pixelHeight;
+        titleAndButtonsContainer.setLayoutParams(layoutParams);
     }
 
     public void setTitleTopMargin(int topMargin) {
-        titleAndButtonsContainer.setTopMargin(topMargin);
+        int marginPx = UiUtils.dpToPx(getContext(), topMargin);
+        ViewGroup.LayoutParams layoutParams = titleAndButtonsContainer.getLayoutParams();
+        MarginLayoutParams marginLp;
+        if (layoutParams instanceof MarginLayoutParams) {
+            marginLp  = (MarginLayoutParams) layoutParams;
+        }else{
+            marginLp = new MarginLayoutParams(layoutParams);
```

Comment:
Why are we creating LayoutParams in a function that sets margin? If `setTitleTopMargin()` is called and there are no layout params we should crash. It's an error, a case that should not happen and should not be supported.

@swabbass Please explain the use case where layout params can be null at this point.
It seems like the `titleAndButtonsContainer` is [created in the constructor](https://github.com/wix/react-native-navigation/blob/5d551cb34ea9d9f607cda888019f3728217d549c/lib/android/app/src/main/java/com/reactnativenavigation/views/stack/topbar/TopBar.java#L71), and immediately added to the hierarchy [here](https://github.com/wix/react-native-navigation/blob/5d551cb34ea9d9f607cda888019f3728217d549c/lib/android/app/src/main/java/com/reactnativenavigation/views/stack/topbar/TopBar.java#L86).

What am I missing? 🤔

Reply:
i think I got what you are talking about, essentially no way that LP for `titleAndButtonsContainer` can be null, since it is being set in its parent c'tor, this is not null check, I'm checking for at least `MarginLayoutParams` as the LP for this view, the other options here for LP is to be `ViewGroup.LayoutParams` (no margin options).


I can make it easier and set `MarginLayoutParams` at c'tor. This will make those checks unnecessary since the across the life span of  `titleAndButtonsContainer`, the LP won't change its type.

Reply:
@swabbass `titleAndButtonsContainer()` is added to a LinearLayout and `LinearLayout.LayoutParams extends ViewGroup.MarginLayoutParams`.

You should assume `titleAndButtonsContainer.layoutParams` always returns MarginLayoutParams. Any check you add around this logic makes the code confusing and misleading.